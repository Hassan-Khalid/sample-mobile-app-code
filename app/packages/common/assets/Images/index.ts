export const COMMON_LOGOS = {
  MAKULA_LOGO_LIGHT: require("./common/makula-logo-light.png"),
  MAKULA_LOGO_DARK: require("./common/makula-logo-dark.png"),
  LOGIN_BG: require("./common/login-bg.png"),
  COMBINE_SHAPE_BG: require("./common/Combined-Shape.png"),
  IC_WARNING: require("./common/ic-warning.png"),
  IC_SEARCH: require("./common/search.png"),
  IC_LIST: require("./common/list.png"),
  IC_PROCEDURE: require("./common/ic-procedure.png"),
  IC_SPARE_PART: require("./common/spare_part.png"),
  IC_SCHEDULE: require("./common/schedule.png"),
  IC_WORK_ORDER: require("./common/folder.png"),
  IC_WORK_ORDER_GREY: require("./common/ic_WO_grey.png"),
  IC_WORK_ORDER_BLUE: require("./common/ic_WO_blue.png"),
  IC_MY_WORK_ORDER_GREY: require("./common/ic_myWO_grey.png"),
  IC_MY_WORK_ORDER_BLUR: require("./common/ic_myWO_Blue.png"),
  IC_USER:require("./common/ic_user.png"),
  IC_CROSS:require("./common/S.png"),
}