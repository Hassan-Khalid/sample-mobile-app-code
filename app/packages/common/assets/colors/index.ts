import Config from 'react-native-config';
import AppColors from './AppColors';

const getAppColors = () => {
  return AppColors;
};

export default getAppColors();
