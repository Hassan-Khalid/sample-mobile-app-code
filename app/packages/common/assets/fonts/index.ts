export const FONT_PROPERTIES = {
  SIZE_8: 8,
  SIZE_10: 10,
  SIZE_12: 12,
  SIZE_14: 14,
  SIZE_16: 16,
  SIZE_18: 18,
  SIZE_20: 20,
  SIZE_22: 22,
  SIZE_28: 28,
  SIZE_34: 34,
  SIZE_60: 60,
  ICON_MEDIUM: 24
};

const APP_FONT = () => {
  return {
    
    
    EXTRA_BOLD: 'Manrope-ExtraBold',
    SEMI_BOLD: 'Manrope-SemiBold',
    BOLD: 'Manrope-Bold',
    MEDIUM: 'Manrope-Medium',
    REGULAR: 'Manrope-Regular',
    LIGHT: 'Manrope-Light',
    EXTRA_LIGHT: 'Manrope-ExtraLight'
  };
};

const APP_FONT_SIZE = () => {
  return {
    HEADER: 20,
    MEDIUM_HEADER: 18,
    SUB_HEADER: 16,
    BODY: 14,
    SUB_BODY: 13,
    CAPTION: 12
  };
};

export { APP_FONT, APP_FONT_SIZE };
