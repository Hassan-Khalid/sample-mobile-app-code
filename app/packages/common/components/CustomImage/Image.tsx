import React from 'react';
import FastImage from 'react-native-fast-image'
import { AnyType } from 'common/declarations/types';
import componentBuilder from 'common/builder/ComponentBuilder';

function Image(props:AnyType) {
  return (
    <FastImage {...props} {...componentBuilder.createTestAttributes('fast-image')}/>
    
  );
}

export default Image;