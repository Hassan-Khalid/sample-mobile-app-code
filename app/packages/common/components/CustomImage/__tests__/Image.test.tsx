import React from 'react';
import { render } from '@testing-library/react-native';
import Image from '../Image';
import { COMMON_LOGOS } from 'common/assets/Images'; // Import the component to be tested

describe('Image component', () => {
  it('renders without crashing', () => {
    // Arrange
    const testProps = {
      source: { uri: 'https://example.com/image.jpg' },
      resizeMode: 'cover',
      style: { width: 100, height: 100 },
    };
    
    // Act
    const { getByTestId } = render(<Image {...testProps} />);
    
    // Assert
    const imageComponent = getByTestId('fast-image'); // Assuming you have set a testID for FastImage
    expect(imageComponent).toBeTruthy(); // Assert that the FastImage component is rendered
  });
  
  it('passes props to FastImage component', () => {
    // Arrange
    const testProps = {
      source:COMMON_LOGOS.LOGIN_BG,
      resizeMode: 'cover',
      style: { width: 100, height: 100 },
    };
    
    // Act
    const  wrapper  = render(<Image {...testProps} />);
    

    // post-conditions
    
    expect(wrapper.toJSON).toMatchSnapshot()
    
  });
});
