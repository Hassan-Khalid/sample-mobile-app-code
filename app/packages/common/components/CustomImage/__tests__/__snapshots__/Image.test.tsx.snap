// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component passes props to FastImage component 1`] = `[Function]`;
